src/script/lua_api/l_util.cpp

```diff
@@ -318,9 +318,16 @@ int ModApiUtil::l_decode_base64(lua_State *L)
 	NO_MAP_LOCK_REQUIRED;
 
 	size_t size;
-	const char *data = luaL_checklstring(L, 1, &size);
+	const char *d = luaL_checklstring(L, 1, &size);
+
+	const std::string data = std::string(d, size);
+
+	if (!base64_is_valid(data)) {
+		lua_pushnil(L);
+		return 1;
```

Comment:
You don't need to explicitly return nil by pushing it, you can instead just `return 0`

Reply:
Thanks, for letting me that know!
Changed it.